app/src/main/java/com/example/weather/ui/forecast/ForecastViewModel.java

```diff
@@ -24,4 +30,18 @@
     private LiveData<List<WeatherData>> fetchAllSavedWeather(){
         return RepositoryImpl.getInstance().getAllWeather();
     }
+
+    @SuppressLint("CheckResult")
+    public void getCurrentWeatherByCoord(double lat, double lon, TextView textView){
+        RepositoryImpl.getInstance().getCurrentWeatherDataByCoord(""+lat,""+lon, "metric")
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Consumer<Float>() {
+                    @SuppressLint("SetTextI18n")
+                    @Override
+                    public void accept(Float aFloat) throws Exception {
+                        textView.setText(""+aFloat);
```

Comment:
We shouldn't change UI directly inside ViewModel. This request should update LiveData, that has observer inside Fragment. Observer inside Fragment should updates our UI on data changed.